src/main_controller.ts

```diff
@@ -381,9 +362,36 @@ export class MainController implements vscode.Disposable {
     private setClientInfo() {
         const versionString = this.extContext.extension.packageJSON.version as string;
         const [major, minor, patch] = [...versionString.split(".").map((n) => +n), 0, 0, 0];
+        logger.debug(`Setting client info: vscode-neovim ${major}.${minor}.${patch}`);
         this.client.setClientInfo("vscode-neovim", { major, minor, patch }, "embedder", {}, {});
     }
 
+    private async setCurrentDir() {
+        let cwd: string | undefined;
+
+        const expectedCwd = vscode.workspace.workspaceFolders?.[0]?.uri.fsPath;
+        if (!expectedCwd) return cwd;
+
+        // Remote Development
+        if (vscode.env.remoteName) {
+            // Runs on the Remote Extension Host
+            if (this.extContext.extension.extensionKind === ExtensionKind.Workspace) {
+                cwd = expectedCwd;
+            }
+        } else {
+            cwd = config.useWsl ? wslpath(expectedCwd) : expectedCwd;
+        }
+
+        if (cwd) {
+            logger.debug(`Setting current dir to: ${cwd}`);
+            try {
+                await this.client.request("nvim_set_current_dir", [cwd]);
```

Comment:
I don't have WSL setup to test with, but why is this different from setting the CWD of the spawned process?

Reply:
When using WSL, we are actually spawning wsl.exe

https://github.com/vscode-neovim/vscode-neovim/blob/4d1ccc93e53631547d4cb89dbf1f29e579775621/src/main_controller.ts#L200-L205